risc0/zkvm/lib.rs

```diff
@@ -12,18 +13,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use anyhow::Result;
+use anyhow::{Error, Result};
 use cxx::{let_cxx_string, UniquePtr};
 
-#[cxx::bridge(namespace = "risc0::rust")]
-mod ffi {
+#[cxx::bridge(namespace = "risc0")]
+pub mod ffi {
     unsafe extern "C++" {
         include!("risc0/zkvm/prove/method_id.h");
 
         type MethodID;
 
-        fn new_method_id(elf_path: &CxxString) -> Result<UniquePtr<MethodID>>;
-        fn write(&self, filename: &CxxString) -> Result<()>;
+        // How do I get this constexpr from C++?
+        //static DIGEST_BYTES: usize;
+
+        fn method_id_from_elf(path: &CxxString) -> Result<UniquePtr<MethodID>>;
+        fn method_id_from_bytes(bytes: &[u8; 384]) -> Result<UniquePtr<MethodID>>;
```

Comment:
This array size exists as a constexpr in C++ (sizeof on the underlying data structure, std::array<ShaDigest, kDigestCount>) but I can't figure out how to get that value in the rust world. Unless someone cleverer than I am figures out how to do that, the options are:

1. redefine the constant on the rust side with a magic number
2. redefine the constant on the rust side by copying all of the underlying constants over and calculating it
3. just use a slice instead of an array, losing some type safety and exposing a bit more of the ffi internals (because slice doesn't have an exact cpp equivalent, so you have to deal with rust::Slice)

thoughts, @flaub? I'm leaning towards #1 as a short term fix and then moving the MethodID definition fully over to rust (with an interface exposed to cpp) as a long term fix, as I don't believe anywhere in c++ world actually cares about the serialization of that thing.

Reply:
We could just use a slice instead of an array, and then just do runtime checking that the len is correct.